src/include/distributed/connection_management.h

```diff
@@ -61,7 +62,14 @@ typedef struct MultiConnection
 	/* time connection establishment was started, for timeout */
 	TimestampTz connectionStart;
 
+	/* membership in ConnectionHashEntry->connections */
 	dlist_node node;
+
+	/* information about the associated remote transaction */
+	RemoteTransaction remoteTransaction;
+
+	/* membership in list of in-progress transactions */
+	dlist_node transactionNode;
```

Comment:
(moving to RemoteTransaction makes access to the connection harder, because you then need to compute the offset to the MultiConnection and do pointer math (i.e. containerof()))